Periodic_4_hyperbolic_triangulation_2/include/CGAL/Periodic_4_hyperbolic_Delaunay_triangulation_traits_2.h

```diff
@@ -0,0 +1,688 @@
+        }
+        else
+        {
+          Euclidean_segment_2* ell2 = boost::get<Euclidean_segment_2>(&s2);
+          Hyperbolic_point_2 p1 = operator()(ell1->supporting_line(), ell2->supporting_line());
+          CGAL_assertion(p1.x()*p1.x() + p1.y()*p1.y()) < FT(1);
+          return p1;
+        }
+      }
+    }
+  };
+
+  Construct_inexact_intersection_2
+  construct_inexact_intersection_2_object() const
+  { return Construct_inexact_intersection_2(); }
+
+  class Construct_inexact_hyperbolic_circumcenter_2
+  {
+    typedef typename Kernel::Circle_2                       Circle_2;
+
+  public:
+    typedef Hyperbolic_Voronoi_point_2                      result_type;
+
+    Hyperbolic_Voronoi_point_2 operator()(const Hyperbolic_point_2& p,
+                                          const Hyperbolic_point_2& q,
+                                          const Hyperbolic_point_2& r)
+    {
+      Origin o;
+      Hyperbolic_point_2 po = Hyperbolic_point_2(o);
+      Circle_2 l_inf(po, FT(1));
+
+      // Check if |p,O| = |q,O| = |r,O| -- then the circumcenter is the origin O
+      if(Compare_distance_2()(po,p,q) == EQUAL && Compare_distance_2()(po,p,r) == EQUAL)
+        return po;
+
+      Euclidean_circle_or_line_2 bis_pq = Construct_circle_or_line_supporting_bisector()(p,q);
+      Euclidean_circle_or_line_2 bis_qr = Construct_circle_or_line_supporting_bisector()(q,r);
+
+      // now supporting objects cannot both be Euclidean lines
+
+      Euclidean_line_2* l;
+      Circle_2* c;
+
+      if(Circle_2* c_pq = boost::get<Circle_2>(&bis_pq))
+      {
+        if(Circle_2* c_qr = boost::get<Circle_2>(&bis_qr))
+        {
+          std::pair<Hyperbolic_point_2, Hyperbolic_point_2> inters = Construct_inexact_intersection_2()(*c_pq, *c_qr);
+
+          if(Has_on_bounded_side_2()(l_inf, inters.first))
+            return inters.first;
+
+          return inters.second;
+        }
+        // here bis_qr is a line
+        l = boost::get<Euclidean_line_2>(&bis_qr);
+        c = c_pq;
+      }
+      else
+      {
+        // here bis_pq is a line
+        l = boost::get<Euclidean_line_2>(&bis_pq);
+        c = boost::get<Circle_2>(&bis_qr);
+      }
+
+      std::pair<Hyperbolic_point_2, Hyperbolic_point_2> inters = Construct_inexact_intersection_2()(*c, *l);
+
+      if(Has_on_bounded_side_2()(l_inf, inters.first))
+        return inters.first;
+
+      return inters.second;
+    }
+
+    template <typename Face_handle>
+    Hyperbolic_Voronoi_point_2 operator()(const Face_handle fh)
+    {
+      return operator()(Construct_hyperbolic_point_2()(fh.vertex(0)->point(), fh.translation(0)),
+                        Construct_hyperbolic_point_2()(fh.vertex(1)->point(), fh.translation(1)),
+                        Construct_hyperbolic_point_2()(fh.vertex(2)->point(), fh.translation(2)) );
+    }
+
+  }; // end Construct_inexact_hyperbolic_circumcenter_2
+
+public:
+  Periodic_4_hyperbolic_Delaunay_triangulation_traits_2() {}
```

Comment:
That constructor should take also be `(const Base& base = Base()) : Base(base)` etc.

Reply:
Done, as for HT2.